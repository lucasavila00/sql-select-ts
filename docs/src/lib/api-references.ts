import { MarkdownDocConfig } from "./types";

export enum ApiReferenceKeys {
  index = "index",
}

const createExamples = <
  T extends { [key in ApiReferenceKeys]: MarkdownDocConfig }
>(
  it: T
): typeof it => it;

export const apiReferenceItems = createExamples({
  [ApiReferenceKeys.index]: {
    contentUrl: new URL(
      `../../../lib/docs/modules/index.ts.md`,
      import.meta.url
    ).toString(),
    title: "Index",
  },
});
