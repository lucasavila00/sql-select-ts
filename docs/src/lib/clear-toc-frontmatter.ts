export const clearTocAndFrontmatter = (str: string): string => {
  const lines = str.split("\n");

  let isIgnoring = false;
  const toRet: string[] = [];

  lines.forEach((line) => {
    if (line.startsWith("---")) {
      isIgnoring = !isIgnoring;
    } else {
      if (!isIgnoring) {
        toRet.push(line);
      }
    }
  });

  return toRet.join("\n").trim();
};
