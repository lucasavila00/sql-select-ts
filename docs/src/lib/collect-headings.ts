import { RenderableTreeNode } from "@markdoc/markdoc";

export type Section = {
  title: string;
  nodeName: string;
  id: string;
};
const collectHeadingsRecursing = (
  node: RenderableTreeNode,
  sections: Section[]
) => {
  if (node != null && typeof node != "string") {
    // Match all h1, h2, h3â€¦ tags
    if (node.name === "HeadingGrommet") {
      const title = node.children[0];

      if (typeof title === "string") {
        sections.push({
          ...node.attributes,
          title,
          nodeName: node.attributes["originalNodeName"],
          id: String(node.attributes["id"]),
        });
      }
    }

    if (node.children) {
      for (const child of node.children) {
        collectHeadingsRecursing(child, sections);
      }
    }
  }
  return sections;
};

export const collectHeadings = (node: RenderableTreeNode): Section[] =>
  collectHeadingsRecursing(node, []);
