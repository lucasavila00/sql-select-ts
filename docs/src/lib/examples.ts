export enum ExamplesKeys {
  select = "select",
  where = "where",
  orderBy = "orderBy",
  join = "join",
  safeString = "safeString",
  typedResponse = "typedResponse",
  limit = "limit",
}

export type ExamplesConfig = { contentUrl: string; title: string };

const createExamples = <T extends { [key in ExamplesKeys]: ExamplesConfig }>(
  it: T
): typeof it => it;

export const examples = createExamples({
  [ExamplesKeys.safeString]: {
    contentUrl: new URL(
      `../../content/safe-string.md`,
      import.meta.url
    ).toString(),
    title: "Safe String",
  },
  [ExamplesKeys.select]: {
    contentUrl: new URL(`../../content/select.md`, import.meta.url).toString(),
    title: "Select",
  },
  [ExamplesKeys.join]: {
    contentUrl: new URL(`../../content/join.md`, import.meta.url).toString(),
    title: "Join",
  },
  [ExamplesKeys.where]: {
    contentUrl: new URL(`../../content/where.md`, import.meta.url).toString(),
    title: "Where",
  },
  [ExamplesKeys.orderBy]: {
    contentUrl: new URL(`../../content/orderBy.md`, import.meta.url).toString(),
    title: "Order By",
  },
  [ExamplesKeys.limit]: {
    contentUrl: new URL(`../../content/limit.md`, import.meta.url).toString(),
    title: "Limit",
  },
  [ExamplesKeys.typedResponse]: {
    contentUrl: new URL(
      `../../content/typed-response.md`,
      import.meta.url
    ).toString(),
    title: "Typed Query Results",
  },
});
