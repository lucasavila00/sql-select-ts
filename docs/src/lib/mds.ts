import { MarkdownDocConfig } from "./types";

export enum MiscMds {
  home = "home",
}

const createExamples = <T extends { [key in MiscMds]: MarkdownDocConfig }>(
  it: T
): typeof it => it;

export const miscMds = createExamples({
  [MiscMds.home]: {
    contentUrl: new URL(`../../mds/home.md`, import.meta.url).toString(),
    title: "sql-select-ts",
  },
});
