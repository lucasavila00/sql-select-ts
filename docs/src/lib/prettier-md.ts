import prettier from "prettier";
import babelParser from "prettier/parser-babel";
import tsParser from "prettier/parser-typescript";
import mdParser from "prettier/parser-markdown";

export const prettyfyMd = (str: string): string =>
  prettier.format(str, {
    filepath: "any.md",
    plugins: [babelParser, tsParser, mdParser],
  });
