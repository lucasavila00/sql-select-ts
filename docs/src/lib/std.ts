export const getEntries = <K extends string | number | symbol, V>(
  obj: Record<K, V>
) => Object.entries(obj) as Array<[K, V]>;

export const getKeys = <T>(obj: T) => Object.keys(obj) as Array<keyof T>;
