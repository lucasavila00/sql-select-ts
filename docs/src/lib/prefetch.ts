import { useEffect } from "react";
import { apiReferenceItems } from "./api-references";
import { examples } from "./examples";
import { miscMds } from "./mds";
import { prefetchMutate } from "./use-document-content";

const prefetchExamples = () =>
  Object.values(examples).forEach((it) => prefetchMutate(it.contentUrl));

const prefetchApiReference = () =>
  Object.values(apiReferenceItems).forEach((it) =>
    prefetchMutate(it.contentUrl)
  );

const prefetchMiscMds = () =>
  Object.values(miscMds).forEach((it) => prefetchMutate(it.contentUrl));

export const usePrefetchAll = () => {
  useEffect(() => {
    prefetchExamples();
    prefetchApiReference();
    prefetchMiscMds();
  }, []);
};
