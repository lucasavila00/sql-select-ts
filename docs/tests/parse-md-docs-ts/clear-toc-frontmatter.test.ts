import * as path from "path";
import * as fs from "fs";
import { clearTocAndFrontmatter } from "../../src/lib/clear-toc-frontmatter";

describe("clear TOC and Frontmatter from docs-ts export", () => {
  it("should clear TOC and Frontmatter from docs-ts export", () => {
    const str = fs.readFileSync(
      path.join(__dirname, "sample-with-toc.md"),
      "utf-8"
    );
    const result = clearTocAndFrontmatter(str);
    expect(result).toMatchSnapshot();
  });
});
