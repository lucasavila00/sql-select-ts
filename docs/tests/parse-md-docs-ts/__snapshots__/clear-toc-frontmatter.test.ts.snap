// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clear TOC and Frontmatter from docs-ts export should clear TOC and Frontmatter from docs-ts export 1`] = `
"## index overview

Entry points of the library.

Added in v0.0.0


# compound

## union

Creates a compound query using 'UNION'

**Signature**

\`\`\`ts
export declare const union: <
  C extends SelectStatement<
    any,
    any,
    any
  >,
  CS extends SelectStatement<
    any,
    any,
    any
  >[]
>(
  content: [C, ...CS]
) => Compound<
  | (C extends SelectStatement<
      infer With,
      infer Scope,
      infer Selection
    >
      ? Selection
      : never)
  | (CS[number] extends SelectStatement<
      infer With,
      infer Scope,
      infer Selection
    >
      ? Selection
      : never),
  C extends SelectStatement<
    infer With,
    infer Scope,
    infer Selection
  >
    ? Selection
    : never
>;
\`\`\`"
`;
